content/guides/gatsby/adding-tina/creating-forms.md

```diff
@@ -0,0 +1,108 @@
+
+```js
+const [modifiedValues, form] = useForm(formConfig)
+```
+
+`modifiedValues` contains the values we provide to the form (inside of `formConfig`) after being modified by the end user. `form` contains the form object created by calling `useForm`, which we'll need in a moment.
+
+## Form Configuration
+
+For details on how to configure forms, take a look at our [form configuration docs](/docs/forms#form-configuration). For the purposes of this guide, we will use the following configuration:
+
+```js
+const formConfig = {
+  id: data.markdownRemark.slug,   // a unique identifier for this instance of the form
+  label: 'Blog Post',             // name of the form to appear in the sidebar
+  initialValues: data,            // populate the form with starting values
+  onSubmit: (values) => {         // do something with the data when the form is submitted
+    alert(`Submitting ${values.frontmatter.title}`)
+  }
+  fields: [                    // define fields to appear in the form
+    {
+      name: 'frontmatter.title',  // field name maps to the corresponding key in initialValues
+      label: 'Title',             // label that appears above the field
+      component: 'text',          // the component used to handle UI and input to the field
+    },
+    {
+      name: 'frontmatter.description',
+      label: 'Description',
+      component: 'textarea',      // `component` accepts a predefined components or a custom React component
+    },
+  ]
+}
+```
+
+> Note that our `onSubmit` handler is just a stub. How you implement this function will depend on how your content is stored, and will be explored in later guides.
+
+## Adding the Form to the Post Component
+
+First, we'll need to import `useForm` and `usePlugin` from the `tinacms` package:
+
+**src/templates/blog-post.js**
+
+```js
+import { useForm, usePlugin } from 'tinacms'
+```
+
+Now, just add the form to the `BlogPostTemplate` component with the configuration we laid out previously. To simplify setup, we'd like to instead give `modifiedValues` the variable name `post` so that our layout code can continue to work without modification:
+
+**src/templates/blog-post.js**
+
+```js
+const BlogPostTemplate = ({ data, pageContext, location }) => {
+   const formConfig = {
+    id: data.markdownRemark.id,
+    label: "Blog Post",
+    initialValues: data.markdownRemark,
+    onSubmit: values => {
+      alert(`Submitting ${values.frontmatter.title}`)
+    },
+    fields: [
+      {
+        name: "frontmatter.title",
+        label: "Title",
+        component: "text",
+      },
+      {
+        name: "frontmatter.description",
+        label: "Description",
+        component: "textarea",
+      },
+    ],
+  }
+  const [post, form] = useForm(formConfig)
+
+  const siteTitle = data.site.siteMetadata.title
+  const { previous, next } = pageContext
```

Comment:
```suggestion

```

Reply:
I feel like these two definitions aren't related / helpful for the example

Reply:
They're part of the demo

Reply:
That is, `gatsby-starter-blog`

Reply:
oh gotcha, okay makes sense